packages/datetime/src/dateInput.tsx

```diff
@@ -275,10 +275,14 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
     }
 
     private handleDateChange = (date: Date, hasUserManuallySelectedDate: boolean) => {
+        const prevMomentDate = this.state.value;
         const momentDate = fromDateToMoment(date);
-        const hasMonthChanged = date !== null && !isMomentNull(this.state.value) && this.state.value.isValid() &&
-            momentDate.month() !== this.state.value.month();
-        const isOpen = !(this.props.closeOnSelection && hasUserManuallySelectedDate && !hasMonthChanged);
+
+        const isOpen = !(this.props.closeOnSelection
```

Comment:
ugh boolean logic. this not-wrapper is throwing my brain for a loop, very hard to follow what's going on here. is there a simpler equivalent expression? #truthtables